actors/builtin/market/market_test.go

```diff
@@ -350,3 +439,135 @@ func (h *marketActorTestHarness) expectProviderControlAddressesAndValidateCaller
 		exitcode.Ok,
 	)
 }
+
+func (h *marketActorTestHarness) withdrawStorageMinerBalanceOK(rt *mock.Runtime, owner, worker, provider address.Address, withDrawAmt, expectedSend abi.TokenAmount) {
+	rt.SetCaller(worker, builtin.AccountActorCodeID)
+	h.expectProviderControlAddressesAndValidateCaller(rt, provider, owner, worker)
+
+	params := market.WithdrawBalanceParams{
+		ProviderOrClientAddress: provider,
+		Amount:                  big.Add(withDrawAmt, big.NewInt(1)),
+	}
+
+	rt.ExpectSend(owner, builtin.MethodSend, nil, expectedSend, nil, exitcode.Ok)
+	rt.Call(h.WithdrawBalance, &params)
+	rt.Verify()
+}
+
+func(h *marketActorTestHarness) withdrawClientBalanceOK(rt *mock.Runtime, client address.Address, withDrawAmt, expectedSend abi.TokenAmount) {
+	rt.SetCaller(client, builtin.AccountActorCodeID)
+	rt.ExpectValidateCallerType(builtin.CallerTypesSignable...)
+	rt.ExpectSend(client, builtin.MethodSend, nil, expectedSend, nil, exitcode.Ok)
+
+	params := market.WithdrawBalanceParams{
+		ProviderOrClientAddress: client,
+		Amount:                  withDrawAmt,
+	}
+
+	rt.Call(h.WithdrawBalance, &params)
+	rt.Verify()
+}
+
+func (h *marketActorTestHarness) publishDealOK(rt *mock.Runtime, deal *market.DealProposal, owner, worker, provider address.Address) abi.DealID {
```

Comment:
We'll very likely need this to be publish deals (plural) for subsequent tests, so just do that now.

Reply:
@anorth I'd like to defer this till we've discussed how to refactor the mock Runtime to allow multiple signature verifications.